### PR TITLE
hotfix/npm-i

### LINKED ISSUE #5: dysfunctional `npm install`
Shout out to maker `@oiclid` for running into this issue and calling it out on the discord.

A couple of issues: 

## `@oiclid`'s experience

When `@oiclid` tried to install the `ryder-cli-proto`, he received the following stack trace:

<details>
<summary>Error: Cannot find module '@oclif/command'</summary>

```shell
node:internal/modules/cjs/loader:943
  throw err;
  ^

Error: Cannot find module '@oclif/command'
Require stack:
- /Users/zerodiv/code/ryder/ryder-cli/bin/run
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:940:15)
    at Function.Module._load (node:internal/modules/cjs/loader:773:27)
    at Module.require (node:internal/modules/cjs/loader:1012:19)
    at require (node:internal/modules/cjs/helpers:93:18)
    at Object.<anonymous> (/Users/zerodiv/code/ryder/ryder-cli/bin/run:3:1)
    at Module._compile (node:internal/modules/cjs/loader:1108:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1137:10)
    at Module.load (node:internal/modules/cjs/loader:988:32)
    at Function.Module._load (node:internal/modules/cjs/loader:828:14)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:76:12) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [ '/Users/zerodiv/code/ryder/ryder-cli/bin/run' ]
}
```
</details>

**Update:** @oiclid finally got it to install with the `-g` option (which is required for now) and then received the following error:

```shell
>  Error: command firmware not found
```

## My experience

And when I went to try and replicate this problem, I uninstalled the CLI via `npm uninstall -g ryder-cli-proto` and uninstalled the `ryderserial-proto` via `npm uninstall -g ryderserial-proto`. Then, to reinstall the CLI, I ran the following command: 

`npm i -g ryder-cli-proto` 

to receive the following error:

```shell
npm install -g https://github.com/Light-Labs/ryder-cli-proto.git
npm ERR! code 127
npm ERR! git dep preparation failed
npm ERR! command /usr/local/Cellar/node/16.0.0_1/bin/node /usr/local/lib/node_modules/npm/bin/npm-cli.js install --force --cache=/Users/chazadmin/.npm --prefer-offline=false --prefer-online=false --offline=false --no-progress --no-save --no-audit
npm ERR! npm WARN using --force Recommended protections disabled.
npm ERR! npm ERR! code 127
npm ERR! npm ERR! path /Users/chazadmin/.npm/_cacache/tmp/git-clone-8001c0fc
npm ERR! npm ERR! command failed
npm ERR! npm ERR! command sh -c npm run prettier && npm run lint && npm run build
npm ERR! npm ERR! > ryderserial-proto@0.0.2 prettier
npm ERR! npm ERR! > prettier --write .
npm ERR! npm ERR! npm WARN using --force Recommended protections disabled.
npm ERR! npm ERR! sh: prettier: command not found
npm ERR! 
npm ERR! npm ERR! A complete log of this run can be found in:
npm ERR! npm ERR!     /Users/chazadmin/.npm/_logs/2021-06-02T12_24_33_452Z-debug.log

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/chazadmin/.npm/_logs/2021-06-02T12_24_34_420Z-debug.log
```

This may be b/c we'd expect someone to already have installed the `ryderserial-proto`? But I personally think the `ryder-cli-proto` should be able to be installed without previous installation of `ryderserial-proto`.

### PR DESCRIPTION
**Pull request by chazkiker2**
Fix lifecycle scripts to build the package (compile TS) on installation or pack

- [x] fix #5 
- [x] include formalization for [NPM publish
](https://www.npmjs.com/package/@lightlabs/ryder-cli-proto)
- [x] version bump
- [x] update `README.md`

### CODE REVIEW

**Comment by chazkiker2** [comment]
@MarvinJanssen this is ready to roll

File: bin/run [hunk 1]
```diff
@@ -1,5 +1,7 @@
 #!/usr/bin/env node
 
-require('@oclif/command').run()
-.then(require('@oclif/command/flush'))
-.catch(require('@oclif/errors/handle'))
+const oclif_command = require("@oclif/command");
+const oclif_cmd_flush = require("@oclif/command/flush");
+const oclif_err_handle = require("@oclif/errors/handle");
+
+oclif_command.run().then(oclif_cmd_flush).catch(oclif_err_handle);
```
Comment (MarvinJanssen) [inline bin/run:7]: Why is this necessary?

**Review by MarvinJanssen — APPROVED** [review]
Amazing work once again!

One idea: maybe we can add oclif readme markers to README.md and have it regenerate?

```markdown
<!-- toc -->
<!-- tocstop -->

# Usage
<!-- usage -->
<!-- usage stop -->

# Commands
<!-- commands -->
<!-- commandsstop -->
```

Then `npx oclif-dev readme`.

Just got to make sure the notes on firmware installs are not lost.

File: bin/run (hunk 1 above)
Comment (chazkiker2) [reply bin/run:7]: totally not necessary; but since there were problems when people were installing from their local clone, I figured this may give them a better error (i.e., `oclif` failed to import properly rather than `oclif` failed to run properly)

can revert if you'd like.

**Comment by chazkiker2** [comment]
> Amazing work once again!
> 
> One idea: maybe we can add oclif readme markers to README.md and have it regenerate?
> 
> ```gfm
> 
> 
> 
> # Usage
> 
> 
> 
> # Commands
> 
> 
> ```
> 
> Then `npx oclif-dev readme`.
> 
> Just got to make sure the notes on firmware installs are not lost.

I like this a lot, didn't know that was a thing! I'll add that in.

File: bin/run (hunk 1 above)
Comment (MarvinJanssen) [reply bin/run:7]: I see, let us just keep it then 👍

**Comment by MarvinJanssen** [comment]
🎉
